### PR TITLE
Handle Namespace/ServiceAccount defaults in PodRunner

### PR DESCRIPTION
**Pull request by tdmanv**
## Pull request type

Please check the type of change your PR introduces:
- [ ] :construction: Work in Progress
- [ ] :rainbow: Refactoring (no functional changes, no api changes)
- [ ] :hamster: Trivial/Minor
- [ ] :bug: Bugfix
- [ ] :sunflower: Feature
- [ ] :world_map: Documentation
- [ ] :robot: Test

## Issues

- #XXX

## Test Plan

- [ ] :muscle: Manual
- [ ] :zap: Unit test
- [ ] :green_heart: E2E

### CODE REVIEW

File: pkg/kube/pod_test.go [hunk 1]
```diff
@@ -62,17 +62,76 @@ func (s *PodSuite) TearDownSuite(c *C) {
...
+			Image:              "kanisterio/kanister-tools:0.28.0",
+			Command:            []string{"sh", "-c", "tail -f /dev/null"},
+			ServiceAccountName: "dummy-sa",
+		},
+		{
+			Namespace:    cns,
+			GenerateName: "test-",
+			Image:        "kanisterio/kanister-tools:0.28.0",
+			Command:      []string{"sh", "-c", "tail -f /dev/null"},
+		},
```
Comment (PrasadG193) [inline pkg/kube/pod_test.go]: Should add one more testcase, where - `Namespace: cns` and `ServiceAccountName: "non-default"`.

**Comment by PrasadG193** [comment]
Overall looks good. Please fix CI failure and suggested test case.

**Comment by viveksinghggits** [comment]
After increasing the timeout the test passes, we can use command 
```
make test
```
to run the tests and the logs can be found [here](https://gist.github.com/viveksinghggits/a59ad8d5e85a18ae5f772bb649720d17).

File: pkg/kube/pod_test.go [hunk 2]
```diff
@@ -62,17 +77,80 @@ func (s *PodSuite) TearDownSuite(c *C) {
...
-	})
-	c.Assert(err, IsNil)
-	c.Assert(WaitForPodReady(ctx, s.cli, s.namespace, pod.Name), IsNil)
-	c.Assert(DeletePod(context.Background(), s.cli, pod), IsNil)
+
+	podOptions := []*PodOptions{
+		{
+			Namespace:    s.namespace,
+			GenerateName: "test-",
+			Image:        "kanisterio/kanister-tools:0.28.0",
```
Comment (tdmanv) [inline pkg/kube/pod_test.go]: please use constants for this image. We should have it defined somewhere

Comment (viveksinghggits) [reply pkg/kube/pod_test.go]: there is one in package function's util.go but 
1. that is not exported and 
2. using that is going to introduce import cycle
so I am planning to make constant in the same source file.

**Pull request merged**